Remotion/Security/UnitTests/ThreadPrincipalProviderTest.cs

```diff
@@ -51,6 +52,29 @@ public void GetUser_NotAuthenticated ()
       Assert.That(_principalProvider.GetPrincipal().IsNull, Is.True);
     }
 
+    [Test]
+    public void GetPrincipal_WithCurrentPrincipalNull_ReturnsNullObject ()
+    {
+      Thread.CurrentPrincipal = null;
+#if NETFRAMEWORK
+      Assert.That(Thread.CurrentPrincipal, Is.Not.Null);
+#else
+      Assert.That(Thread.CurrentPrincipal, Is.Null);
+#endif
+      Assert.That(_principalProvider.GetPrincipal().IsNull, Is.True);
```

Comment:
Interesting, .NET Framework is actually forcing a non-null value into the Thread.CurrentPrincipal when you try to set it to null for tests reasons (unless you also override the AppDomain's PrincipalPolicy).